src/main/java/com/synopsys/integration/alert/channel/jira/server/JiraServerIssueConfigValidator.java

```diff
@@ -0,0 +1,161 @@
+        IssueConfig jiraIssueConfig = new IssueConfig();
+        Map<String, String> fieldErrors = new HashMap<>();
+
+        ProjectComponent projectComponent = validateProject(fieldAccessor, fieldErrors);
+        if (projectComponent != null) {
+            jiraIssueConfig.setProjectId(projectComponent.getId());
+            jiraIssueConfig.setProjectKey(projectComponent.getKey());
+            jiraIssueConfig.setProjectName(projectComponent.getName());
+        }
+
+        String issueCreator = validateIssueCreator(fieldAccessor, fieldErrors);
+        jiraIssueConfig.setIssueCreator(issueCreator);
+
+        String issueType = validateIssueType(fieldAccessor, fieldErrors);
+        jiraIssueConfig.setIssueType(issueType);
+
+        Boolean commentOnIssue = fieldAccessor.getBooleanOrFalse(JiraServerDescriptor.KEY_ADD_COMMENTS);
+        jiraIssueConfig.setCommentOnIssues(commentOnIssue);
+
+        String resolveTransition = fieldAccessor.getStringOrNull(JiraServerDescriptor.KEY_RESOLVE_WORKFLOW_TRANSITION);
+        jiraIssueConfig.setResolveTransition(resolveTransition);
+
+        String openTransition = fieldAccessor.getStringOrNull(JiraServerDescriptor.KEY_OPEN_WORKFLOW_TRANSITION);
+        jiraIssueConfig.setOpenTransition(openTransition);
+
+        if (fieldErrors.isEmpty()) {
+            return jiraIssueConfig;
+        } else {
+            throw new AlertFieldException(fieldErrors);
+        }
+    }
+
+    private ProjectComponent validateProject(FieldAccessor fieldAccessor, Map<String, String> fieldErrors) {
+        Optional<String> optionalProjectName = fieldAccessor.getString(JiraServerDescriptor.KEY_JIRA_PROJECT_NAME);
+        if (optionalProjectName.isPresent()) {
+            String jiraProjectName = optionalProjectName.get();
+            try {
+                List<ProjectComponent> projectsResponseModel = projectService.getProjectsByName(jiraProjectName);
+                Optional<ProjectComponent> optionalProject = projectsResponseModel
+                                                                 .stream()
+                                                                 .filter(project -> jiraProjectName.equals(project.getName()) || jiraProjectName.equals(project.getKey()))
+                                                                 .findAny();
+                if (optionalProject.isPresent()) {
+                    return optionalProject.get();
+                } else {
+                    fieldErrors.put(JiraServerDescriptor.KEY_JIRA_PROJECT_NAME, String.format("No project named '%s' was found", jiraProjectName));
+                }
+            } catch (IntegrationException e) {
+                fieldErrors.put(JiraServerDescriptor.KEY_JIRA_PROJECT_NAME, String.format(CONNECTION_ERROR_FORMAT_STRING, "projects"));
+            }
+        } else {
+            requireField(fieldErrors, JiraServerDescriptor.KEY_JIRA_PROJECT_NAME);
+        }
+        return null;
```

Comment:
Should the return Optional.empty instead or is it fine returning null since it's private? Same thing for the other private methods in this class.